src/main/java/com/enderio/core/common/transform/EnderCoreTransformer.java

```diff
@@ -28,14 +31,59 @@
 
 @MCVersion(value = "1.7.10")
 public class EnderCoreTransformer implements IClassTransformer {
+
+  // The EnderCore class cannot be referenced from the transformer,
+  // as it improperly triggers classloading of FML event classes.
+  // This is caused by its various event handlers. Since event handlers
+  // require an event class to be referenced in the method signature
+  // (in this case, a state event), the referenced event class is
+  // loaded along with the containing class of the method.
+  // Since EnderCore is referenced in the 'transform' method of this class,
+  // the aforementioned event classes are consequently referenced and classloaded too early.
+  //
+  // In order to be compatible with coremods such as Sponge, FML event
+  // classes cannot be classloaded until the game is starting. Specifically,
+  // there are being loaded before FML is ready to actually start posting state events.
```

Comment:
That's a good point - I didn't realize that 'loading' could be ambiguous when dealing with mods. I'll change it